pkg/statsd/handler_backend.go

```diff
@@ -194,10 +194,12 @@ func (bh *BackendHandler) DispatchEvent(ctx context.Context, e *gostatsd.Event)
 		case bh.concurrentEvents <- struct{}{}:
 			// Creates a new context for dispatching the event.
 			// We create a new one otherwise it uses the request context which is cancelled as soon as this function returns.
-			//nolint:govet
-			timeoutCtx, _ := context.WithTimeout(context.Background(), 20*time.Second)
-			go bh.internalDispatchEvent(timeoutCtx, backend, e)
-			eventsDispatched++
+			go func() {
+				timeoutCtx, cancelTimeout := context.WithTimeout(context.Background(), 20*time.Second)
+				defer cancelTimeout()
+				bh.internalDispatchEvent(timeoutCtx, backend, e)
+				eventsDispatched++
```

Comment:
This is now a race condition, `eventsDispatched` needs to be outside the go func.  `eventsDispatched` is a variable from the outer scope, being captured by the func (ie, it's storing a reference to it).  This means if there's multiple backends setup, then two goroutines could both try to update it, without any kind of synchronization.

I also just realized there's also a similar hazard with `backend`, except from the other direction.  The loop modifies `backend`.  The function will always be referring to the outer variable, not its own copy.  You can see the behavior at https://play.golang.org/p/g6jPXwxeXw0